src/permissions_template.c

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+
+#include "kerneltypes.h"
+#include "kmalloc.h"
+#include "permissions_template.h"
+#include "console.h"
+
+#define MAX_NUMBER_OF_PERMISSIONS 169 // temporary, for now they'll be statically allocated
+#define DEFAULT_FRACTION_OF_MEMORY 3
+
+static uint32_t template_counter = 1;
```

Comment:
The postfix returns the original value, so the first identifier will be 1. This is intentional; see my response to Alex's comment